src/rsz/src/RepairDesign.cc

```diff
@@ -391,92 +387,47 @@ RepairDesign::repairNet(Net *net,
-            if (drvr_port) {
-              float max_cap1 = findSlewLoadCap(drvr_port, max_slew1, corner1);
-              max_cap = min(max_cap, max_cap1);
-            }
-            corner = corner1;
-            if (slew_slack1 < 0.0) {
-              debugPrint(logger_, RSZ, "repair_net", 2,
-                         "drvr slew violation slew={} max_slew={}",
-                         delayAsString(slew1, this, 3),
-                         delayAsString(max_slew1, this, 3));
-              repair_slew = true;
-              slew_violations++;
-            }
-          }
-          // Check slew at the loads.
-          // Note that many liberty libraries do not have max_transition attributes on
-          // input pins.
-          // Max slew violations at the load pins are repaired by inserting buffers
-          // and reducing the wire length to the load.
-          resizer_->checkLoadSlews(drvr_pin, slew_margin_,
-                                   slew1, max_slew1, slew_slack1, corner1);
-          if (slew_slack1 < 0.0) {
-            debugPrint(logger_, RSZ, "repair_net", 2,
-                       "load slew violation load_slew={} max_slew={}",
-                       delayAsString(slew1, this, 3),
-                       delayAsString(max_slew1, this, 3));
-            corner = corner1;
-            // Don't double count the driver/load on same net.
-            if (!repair_slew) {
-              slew_violations++;
-            }
-            repair_slew = true;
-          }
-        }
-        if (repair_slew
-            || repair_cap
-            || repair_wire) {
-          Point drvr_loc = db_network_->location(drvr->pin());
-          debugPrint(logger_, RSZ, "repair_net", 1, "driver {} ({} {}) l={}",
-                     sdc_network_->pathName(drvr_pin),
-                     units_->distanceUnit()->asString(dbuToMeters(drvr_loc.getX()), 1),
-                     units_->distanceUnit()->asString(dbuToMeters(drvr_loc.getY()), 1),
-                     units_->distanceUnit()->asString(dbuToMeters(wire_length), 1));
-          repairNet(bnet, drvr_pin, max_cap, max_length, corner);
-          repaired_net = true;
-
+        bool need_repair = needRepair(drvr_pin,
+                                      corner,
+                                      max_length,
+                                      wire_length,
+                                      check_cap,
+                                      check_slew,
+                                      max_cap,
+                                      slew_violations,
+                                      cap_violations,
+                                      length_violations);
+        
+        if (need_repair) {
           if (resize_drvr) {
             resize_count_ += resizer_->resizeToTargetSlew(drvr_pin);
           }
+          wire_length = bnet->maxLoadWireLength();
+          need_repair = needRepair(drvr_pin,
+                                   corner,
```

Comment:
If we don't do resizeToTargetSlew there is no need to redo needRepair so this call can be moved inside the if.